src/modules/mc_pos_control/MulticopterPositionControl.cpp

```diff
@@ -288,36 +257,33 @@ void MulticopterPositionControl::Run()
 			// in stabilized mode attitude setpoints get ignored
 			// in offboard with attitude setpoints they come from MAVLink directly
 			vehicle_attitude_setpoint_s attitude_setpoint{};
-			attitude_setpoint.timestamp = time_stamp_now;
 			_control.getAttitudeSetpoint(attitude_setpoint);
+			attitude_setpoint.timestamp = hrt_absolute_time();
 			_vehicle_attitude_setpoint_pub.publish(attitude_setpoint);
-
-		} else {
-			// reset the numerical derivatives to not generate d term spikes when coming from non-position controlled operation
-			_vel_x_deriv.reset();
-			_vel_y_deriv.reset();
-			_vel_z_deriv.reset();
 		}
+
+	} else {
+		_time_stamp_last_loop = 0;
```

Comment:
Aha, that's how you catch smaller than 500_ms but still significant toggles of the climb rate flag. Newbie scheduling question: Couldn't it ever be that `Run()` is called but there's no `vehicle_local_position` subscription update? That would immediately reset the derivative which might be desired depending on the case.